pkg/scheduler/queue/multi_queuing_algorithm_tree_queue_benchmark_test.go

```diff
@@ -0,0 +1,533 @@
+				},
+			},
+		},
+		{
+			name: "2 tenants, first with 25pct slow queries, second with 75pct slow queries",
+			tenantQueueDimensionsWeights: map[string]map[string]float64{
+				"0": {
+					ingesterQueueDimension:                .75,
+					storeGatewayQueueDimension:            .125,
+					ingesterAndStoreGatewayQueueDimension: .125,
+				},
+				"1": {
+					ingesterQueueDimension:                .25,
+					storeGatewayQueueDimension:            .375,
+					ingesterAndStoreGatewayQueueDimension: .375,
+				},
+			},
+		},
+		{
+			name: "2 tenants, first with 50pct slow queries, second with 50pct slow queries",
+			tenantQueueDimensionsWeights: map[string]map[string]float64{
+				"0": {
+					ingesterQueueDimension:                .50,
+					storeGatewayQueueDimension:            .25,
+					ingesterAndStoreGatewayQueueDimension: .25,
+				},
+				"1": {
+					ingesterQueueDimension:                .50,
+					storeGatewayQueueDimension:            .25,
+					ingesterAndStoreGatewayQueueDimension: .25,
+				},
+			},
+		},
+	}
+
+	maxQueriersPerTenant := 0 // disable shuffle sharding
+
+	// Increase totalRequests to tighten up variations when running locally, but do not commit higher values;
+	// the later test cases with a higher percentage of slow queries will take a long time to run.
+	totalRequests := 1000
+	numProducers := 10
+	// Number of consumers must be divisible by the number of query component queue types
+	// for even distribution of workers to queues and fair comparison between algorithms.
+	// Query component queue types can up to 4: ingester, store-gateway, ingester-and-store-gateway, and unknown.
+	// The unknown queue type is not used in this test, but 12 ensures divisibility by 4, 3, 2, and 1.
+	numConsumers := 12
+
+	normalConsumerLatency := 1 * time.Millisecond
+	// slow request approximately 100x longer than the fast request seems fair;
+	// an ingester can respond in 0.3 seconds while a slow store-gateway query can take 30 seconds
+	slowConsumerLatency := 100 * time.Millisecond
+
+	var testCaseNames []string
+	testCaseReports := map[string]*testScenarioQueueDurationReport{}
+
+	for _, weightedQueueDimensionTestCase := range weightedQueueDimensionTestCases {
+		numTenants := len(weightedQueueDimensionTestCase.tenantQueueDimensionsWeights)
+
+		tqa := newTenantQuerierAssignments(0)
+
+		nonFlippedRoundRobinTree, err := NewTree(tqa, &roundRobinState{}, &roundRobinState{})
+		require.NoError(t, err)
+
+		flippedRoundRobinTree, err := NewTree(&roundRobinState{}, tqa, &roundRobinState{})
+		require.NoError(t, err)
+
+		querierWorkerPrioritizationTree, err := NewTree(NewQuerierWorkerQueuePriorityAlgo(), tqa, &roundRobinState{})
+		require.NoError(t, err)
+
+		trees := []struct {
+			name string
+			tree Tree
+		}{
+			// keeping these names the same length keeps logged results aligned
+			{
+				"non-flipped round-robin tree",
+				nonFlippedRoundRobinTree,
+			},
+			{
+				"new flipped round-robin tree",
```

Comment:
```suggestion
				"flipped round-robin tree",
```

Reply:
:laughing: I hear you.. but you know how much time I have spent trying to make the names both somewhat accurate but also the same length? The output is a bear to read if we don't line everything up and I preferred not to just pad with extra characters.

Reply:
after some effort I think I got better descriptions while managing to keep the same length:

```golang
{
	"tenant-querier -> query component round-robin tree",
	nonFlippedRoundRobinTree,
},
{
	"tenant-querier -> query component round-robin tree",
	flippedRoundRobinTree,
},
{
	"worker-queue prioritization -> tenant-querier tree",
	querierWorkerPrioritizationTree,
},
```